prov/rxm/src/rxm_ep.c

```diff
@@ -1393,84 +1270,85 @@ rxm_ep_inject_common(struct rxm_ep *rxm_ep, const void *buf, size_t len,
 	}
 
 	if (pkt_size <= rxm_ep->msg_info->tx_attr->inject_size) {
-		struct rxm_tx_buf *tx_buf;
-		ret = rxm_ep_format_tx_inject_buf(rxm_ep, rxm_conn, buf,
-						  len, data, flags, tag,
-						  op, comp_flags, &tx_buf);
-		if (OFI_UNLIKELY(ret))
-	    		return ret;
+		struct rxm_tx_base_buf *tx_buf = (struct rxm_tx_base_buf *)
+			rxm_tx_buf_get(rxm_ep, RXM_BUF_POOL_TX_INJECT);
```

Comment:
Do we need `RXM_BUF_POOL_TX_INJECT` pool? Since the message is sent as fi_inject, we can just use some buffer from the stack.

Reply:
It can lead to stack overflow if underlying provider reports too large value in `fi_info::tx_attr::inject_size`